File: frontend/lib/core/api/api_endpoints.dart
/// Backend API endpointlari.
/// Manba: API_DOCS.md. Base URL `.env` / `--dart-define=API_URL` orqali keladi
/// (qarang: [ApiEndpoints.baseUrl]).
abstract class ApiEndpoints {
  /// Dev uchun standart. Prod build'da --dart-define=API_URL bilan almashtiriladi.
  static const baseUrl = String.fromEnvironment(
    'API_URL',
    defaultValue: 'http://localhost:4000/api/v1',
  );

  static const wsUrl = String.fromEnvironment(
    'WS_URL',
    defaultValue: 'http://localhost:4000',
  );

  // ---- Auth ----
  static const register = '/auth/register';
  static const verifyEmail = '/auth/verify-email';
  static const resendVerification = '/auth/resend-verification';
  static const login = '/auth/login';
  static const twoFaVerifyLogin = '/auth/2fa/verify-login';
  static const refresh = '/auth/refresh';
  static const logout = '/auth/logout';
  static const me = '/auth/me';
  static const changePassword = '/auth/change-password';
  static const forgotPassword = '/auth/forgot-password';
  static const resetPassword = '/auth/reset-password';
  static const twoFaSetup = '/auth/2fa/setup';
  static const twoFaEnable = '/auth/2fa/enable';
  static const twoFaDisable = '/auth/2fa/disable';

  // ---- Users ----
  static const usersMe = '/users/me';
  static const usersSearch = '/users/search';
  static String userById(String id) => '/users/$id';

  // ---- Posts ----
  static const postsFeed = '/posts/feed';
  static const posts = '/posts';
  static String postById(String id) => '/posts/$id';
  static String postLike(String id) => '/posts/$id/like';
  static String postLiked(String id) => '/posts/$id/liked';
  static String postComments(String id) => '/posts/$id/comments';
  static String postComment(String id, String commentId) =>
      '/posts/$id/comments/$commentId';
  static String postPin(String id) => '/posts/$id/pin';

  // ---- Messages (REST) ----
  static String messages(String chatType, String chatId) =>
      '/messages/$chatType/$chatId';
  static String messageById(String id) => '/messages/$id';
  static String messageReact(String id) => '/messages/$id/react';
  static String messageReadBy(String id) => '/messages/$id/read-by';

  // ---- Groups ----
  static const groups = '/groups';
  static const groupsMy = '/groups/my';
  static String groupById(String id) => '/groups/$id';
  static String groupJoin(String id) => '/groups/$id/join';
  static String groupJoinByCode(String code) => '/groups/join/$code';
  static String groupLeave(String id) => '/groups/$id/leave';
  static String groupMembers(String id) => '/groups/$id/members';
  static String groupMember(String id, String userId) =>
      '/groups/$id/members/$userId';

  // ---- Channels ----
  static const channels = '/channels';
  static const channelsMy = '/channels/my';
  static String channelById(String id) => '/channels/$id';
  static String channelSubscribe(String id) => '/channels/$id/subscribe';
  static String channelUnsubscribe(String id) => '/channels/$id/unsubscribe';
  static String channelSubscribers(String id) => '/channels/$id/subscribers';
  static String channelStats(String id) => '/channels/$id/stats';

  // ---- Goals ----
  static const goals = '/goals';
  static const goalsTrending = '/goals/trending';
  static const goalsMy = '/goals/my';
  static String goalById(String id) => '/goals/$id';
  static String goalJoin(String id) => '/goals/$id/join';
  static String goalLeave(String id) => '/goals/$id/leave';

  // ---- Calls (REST) ----
  static const callInitiate = '/calls/initiate';
  static String callJoin(String id) => '/calls/$id/join';
  static String callLeave(String id) => '/calls/$id/leave';
  static String callEnd(String id) => '/calls/$id/end';
  static String callParticipants(String id) => '/calls/$id/participants';
  static const callHistory = '/calls/history';

  // ---- Notifications ----
  static const notifications = '/notifications';
  static const notificationsReadAll = '/notifications/read-all';
  static String notificationRead(String id) => '/notifications/$id/read';
  static String notificationById(String id) => '/notifications/$id';
  static const notificationsPushRegister = '/notifications/push/register';

  // ---- Search ----
  static const search = '/search';
  static const searchSuggestions = '/search/suggestions';

  // ---- Media ----
  static const mediaUpload = '/media/upload';
  static String mediaByKey(String key) => '/media/$key';
  static String mediaPresigned(String key) => '/media/presigned/$key';

  // ---- Health ----
  static const health = '/health';

  // ---- WebSocket namespaces ----
  static const wsChatNamespace = '/chat';
  static const wsCallsNamespace = '/calls';
  static const wsNotificationsNamespace = '/notifications';
}

File: frontend/src/app/core/services/api/api-endpoints.ts
import { environment } from '../../../../environments/environment';

export const API_BASE_URL = environment.apiUrl;

export const ApiEndpoints = {
  auth: {
    register: `${API_BASE_URL}/auth/register`,
    verifyEmail: `${API_BASE_URL}/auth/verify-email`,
    resendVerification: `${API_BASE_URL}/auth/resend-verification`,
    login: `${API_BASE_URL}/auth/login`,
    verify2FaLogin: `${API_BASE_URL}/auth/2fa/verify-login`,
    refresh: `${API_BASE_URL}/auth/refresh`,
    logout: `${API_BASE_URL}/auth/logout`,
    me: `${API_BASE_URL}/auth/me`,
    changePassword: `${API_BASE_URL}/auth/change-password`,
    forgotPassword: `${API_BASE_URL}/auth/forgot-password`,
    resetPassword: `${API_BASE_URL}/auth/reset-password`,
    twoFaSetup: `${API_BASE_URL}/auth/2fa/setup`,
    twoFaEnable: `${API_BASE_URL}/auth/2fa/enable`,
    twoFaDisable: `${API_BASE_URL}/auth/2fa/disable`,
  },
  users: {
    me: `${API_BASE_URL}/users/me`,
    search: `${API_BASE_URL}/users/search`,
    byId: (id: string) => `${API_BASE_URL}/users/${id}`,
  },
  goals: {
    root: `${API_BASE_URL}/goals`,
    trending: `${API_BASE_URL}/goals/trending`,
    my: `${API_BASE_URL}/goals/my`,
    byId: (id: string) => `${API_BASE_URL}/goals/${id}`,
    join: (id: string) => `${API_BASE_URL}/goals/${id}/join`,
    leave: (id: string) => `${API_BASE_URL}/goals/${id}/leave`,
  },
  groups: {
    root: `${API_BASE_URL}/groups`,
    my: `${API_BASE_URL}/groups/my`,
    byId: (id: string) => `${API_BASE_URL}/groups/${id}`,
    join: (id: string) => `${API_BASE_URL}/groups/${id}/join`,
    joinByCode: (code: string) => `${API_BASE_URL}/groups/join/${code}`,
    leave: (id: string) => `${API_BASE_URL}/groups/${id}/leave`,
    members: (id: string) => `${API_BASE_URL}/groups/${id}/members`,
    member: (id: string, userId: string) => `${API_BASE_URL}/groups/${id}/members/${userId}`,
  },
  channels: {
    root: `${API_BASE_URL}/channels`,
    my: `${API_BASE_URL}/channels/my`,
    byId: (id: string) => `${API_BASE_URL}/channels/${id}`,
    subscribe: (id: string) => `${API_BASE_URL}/channels/${id}/subscribe`,
    unsubscribe: (id: string) => `${API_BASE_URL}/channels/${id}/unsubscribe`,
    subscribers: (id: string) => `${API_BASE_URL}/channels/${id}/subscribers`,
    stats: (id: string) => `${API_BASE_URL}/channels/${id}/stats`,
  },
  messages: {
    byChat: (chatType: string, chatId: string) =>
      `${API_BASE_URL}/messages/${chatType}/${chatId}`,
    byId: (id: string) => `${API_BASE_URL}/messages/${id}`,
    react: (id: string) => `${API_BASE_URL}/messages/${id}/react`,
    readBy: (id: string) => `${API_BASE_URL}/messages/${id}/read-by`,
  },
  posts: {
    root: `${API_BASE_URL}/posts`,
    feed: `${API_BASE_URL}/posts/feed`,
    byId: (id: string) => `${API_BASE_URL}/posts/${id}`,
    like: (id: string) => `${API_BASE_URL}/posts/${id}/like`,
    liked: (id: string) => `${API_BASE_URL}/posts/${id}/liked`,
    pin: (id: string) => `${API_BASE_URL}/posts/${id}/pin`,
    comments: (id: string) => `${API_BASE_URL}/posts/${id}/comments`,
    comment: (id: string, commentId: string) =>
      `${API_BASE_URL}/posts/${id}/comments/${commentId}`,
  },
  notifications: {
    root: `${API_BASE_URL}/notifications`,
    unreadCount: `${API_BASE_URL}/notifications/unread-count`,
    readAll: `${API_BASE_URL}/notifications/read-all`,
    read: (id: string) => `${API_BASE_URL}/notifications/${id}/read`,
    byId: (id: string) => `${API_BASE_URL}/notifications/${id}`,
    pushRegister: `${API_BASE_URL}/notifications/push/register`,
  },
  calls: {
    initiate: `${API_BASE_URL}/calls/initiate`,
    active: `${API_BASE_URL}/calls/active`,
    history: `${API_BASE_URL}/calls/history`,
    join: (id: string) => `${API_BASE_URL}/calls/${id}/join`,
    leave: (id: string) => `${API_BASE_URL}/calls/${id}/leave`,
    end: (id: string) => `${API_BASE_URL}/calls/${id}/end`,
    participants: (id: string) => `${API_BASE_URL}/calls/${id}/participants`,
  },
  media: {
    upload: `${API_BASE_URL}/media/upload`,
    byKey: (key: string) => `${API_BASE_URL}/media/${key}`,
    presigned: (key: string) => `${API_BASE_URL}/media/presigned/${key}`,
  },
} as const;

export const AUTH_SKIP_URLS = [
  ApiEndpoints.auth.login,
  ApiEndpoints.auth.register,
  ApiEndpoints.auth.refresh,
  ApiEndpoints.auth.verifyEmail,
  ApiEndpoints.auth.resendVerification,
  ApiEndpoints.auth.forgotPassword,
  ApiEndpoints.auth.resetPassword,
];
